tree/dataframe/test/dataframe_merge_results.cxx

```diff
@@ -0,0 +1,172 @@
+#include "ROOT/RDataFrame.hxx"
+#include "ROOT/RResultPtr.hxx"
+#include "gtest/gtest.h"
+
+TEST(RDataFrameMergeResults, MergeCount)
+{
+    ROOT::RDataFrame df{100};
+    auto mycount = df.Count();
+
+    auto mergeablecount = ROOT::Detail::RDF::GetMergeableValue(mycount);
+    auto othermergeablecount = ROOT::Detail::RDF::GetMergeableValue(mycount);
+
+    mergeablecount->Merge(othermergeablecount);
+    auto mergedcount = mergeablecount->GetValue();
+
+    EXPECT_EQ(mergedcount, 200);
+}
+
+TEST(RDataFrameMergeResults, MergeMean)
+{
+    ROOT::RDataFrame df1{10};
+    ROOT::RDataFrame df2{20};
+    
+    auto col1 = df1.Define("x","double(rdfentry_)");
+    auto col2 = df2.Define("x","double(rdfentry_)");
+
+    auto mean1 = col1.Mean("x");
+    auto mean2 = col2.Mean("x");
```

Comment:
these unit tests are executed _many_ times per day. In order to speed them up a bit, please specify template parameters to all RDF actions so we don't have to go through jitting. e.g. `col1.Mean<double>` or `col1.Histo1D<double>`, and use lambdas rather than strings as `Define` expressions

Reply:
done